tests/utilities/test_plugins.py

```diff
@@ -0,0 +1,60 @@
+import inspect
+import shutil
+import unittest
+from pathlib import Path
+from unittest.mock import patch
+import os
+import importlib.util
+
+import jinja2
+
+from dagger.utilities.module import Module  # Adjust this import according to your actual module structure
+from dagger import conf
+
+TESTS_ROOT = Path(__file__).parent.parent
+
+class TestLoadPlugins(unittest.TestCase):
+
+    def setUp(self):
+        self._jinja_environment = jinja2.Environment()
+        loaded_classes = Module.load_plugins()
+        for class_name, class_obj in loaded_classes.items():
+            self._jinja_environment.globals[class_name] = class_obj
+
+        self._template = self._jinja_environment.from_string("inputs: {{ SampleFolderPlugin.get_inputs() }}")
+
+    @patch("dagger.conf.PLUGIN_DIRS", new=[])
+    @patch("os.walk")
+    def test_load_plugins_no_plugin_dir(self, mock_os_walk):
+        # Simulate os.walk returning no Python files
+        mock_os_walk.return_value = [("/fake/plugin/dir", [], [])]
+
+        result = Module.load_plugins()
+
+        # Expecting an empty dictionary since no plugins were found
+        self.assertEqual(result, {})
+
+    @patch("dagger.conf.PLUGIN_DIRS", new=[str(TESTS_ROOT.joinpath("fixtures/plugins"))])
+    def test_load_plugins(self):
+
+        result = Module.load_plugins()
+        for name, plugin_class in result.items():
+            result[name] = str(plugin_class)
```

Comment:
The same here

Reply:
this part is not in the actual code. 

it existed in the tests because it was hard to compare if an instance of a `class` was returned. So, i converted the value of the dictionary to string, just for the test.

but i changed this now, and it is not part of the test anymore